MoreLinq.Test/SequenceTest.cs

```diff
@@ -15,80 +15,129 @@
 // limitations under the License.
 #endregion
 
-using NUnit.Framework;
-using System;
-using System.Linq;
-
 namespace MoreLinq.Test
 {
+    using NUnit.Framework;
+    using System;
+    using System.Linq;
+
     [TestFixture]
     public class SequenceTest
     {
+        [TestCase(-10, -4)]
+        [TestCase(-1, 5)]
         [TestCase(1, 10)]
         [TestCase(30, 55)]
         [TestCase(27, 172)]
-        public void SequenceAscendingTest(int start, int stop)
+        public void SequenceWithAscendingRange(int start, int stop)
         {
             var result = MoreEnumerable.Sequence(start, stop);
             var expectations = Enumerable.Range(start, stop - start + 1);
 
             Assert.That(result, Is.EqualTo(expectations));
         }
 
+        [TestCase(-4, -10)]
+        [TestCase(5, -1)]
         [TestCase(10, 1)]
         [TestCase(55, 30)]
         [TestCase(172, 27)]
-        public void SequenceDescendingTest(int start, int stop)
+        public void SequenceWithDescendingRange(int start, int stop)
         {
             var result = MoreEnumerable.Sequence(start, stop);
             var expectations = Enumerable.Range(stop, start - stop + 1).Reverse();
 
             Assert.That(result, Is.EqualTo(expectations));
         }
 
-
+        [TestCase(-10, -4, 2)]
+        [TestCase(-1, 5, 3)]
         [TestCase(1, 10, 1)]
         [TestCase(30, 55, 4)]
         [TestCase(27, 172, 9)]
-        public void SequenceAscendingWithAscendingStepTest(int start, int stop, int step)
+        public void SequenceWithAscendingRangeWithAscendingStep(int start, int stop, int step)
```

Comment:
`SequenceWithAscendingRangeWithAscendingStep` &rarr; `SequenceWithAscendingRangeAscendingStep`

Second `With` is redundant.